api-2.2/src/main/java/org/openmrs/module/coreapps/htmlformentry/EncounterDiagnosesElement.java

```diff
@@ -209,26 +216,44 @@ public void handleSubmission(FormEntrySession formEntrySession, HttpServletReque
 
         try {
             String jsonList = request.getParameter("encounterDiagnoses");
+
             JsonNode list = new ObjectMapper().readTree(jsonList);
+
             for (JsonNode node : list) {
                 CodedOrFreeTextAnswer answer = new CodedOrFreeTextAnswer(node.get("diagnosis").getTextValue(), conceptService);
                 Diagnosis.Order diagnosisOrder = Diagnosis.Order.valueOf(node.get("order").getTextValue());
                 Diagnosis.Certainty certainty = Diagnosis.Certainty.valueOf(node.get("certainty").getTextValue());
+                Integer existingDiagnosis = null;
 
                 if (node.path("existingObs").getNumberValue() != null) {
                     JsonNode nd = node.get("existingObs");
                     System.out.println(nd);
                 }
 
-                org.openmrs.Diagnosis diagnosis = new org.openmrs.Diagnosis();
-                diagnosis.setEncounter(formEntrySession.getEncounter());
-                diagnosis.setCertainty(certainty == Diagnosis.Certainty.CONFIRMED ? ConditionVerificationStatus.CONFIRMED : ConditionVerificationStatus.PROVISIONAL);
-                diagnosis.setRank(diagnosisOrder == Diagnosis.Order.PRIMARY ? 1 : 2);
-                diagnosis.setPatient(formEntrySession.getPatient());
-                diagnosis.setDiagnosis(new CodedOrFreeText(answer.getCodedAnswer(), answer.getSpecificCodedAnswer(), answer.getNonCodedAnswer()));
+                if (node.path("existingDiagnosis").getNumberValue() != null) {
+                    existingDiagnosis = node.get("existingDiagnosis").getIntValue();
+                }
+
+                org.openmrs.Diagnosis diagnosis;
+
+                if(existingDiagnosis !=null){
+                    diagnosis = Context.getDiagnosisService().getDiagnosis(existingDiagnosis);
+                    diagnosis.setDiagnosis(Context.getDiagnosisService().getDiagnosis(existingDiagnosis).getDiagnosis());
+                    diagnosis.setDateChanged(new Date());
```

Comment:
Can you save an edited diagnosis only if any of its properties has been changed?